src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/resource/SmokeTestResource.kt

```diff
@@ -66,7 +66,7 @@ class SmokeTestResource(
   ): Flux<TestStatus> = runCatching { PsiTestProfiles.valueOf(testProfile).profile }
     .map { profile ->
       psiSmokeTestService.runSmokeTest(profile).concatMap {
-        psiSmokeTestService.cleanup(profile).thenReturn(it)
+        if (it.hasResult()) psiSmokeTestService.cleanup(profile).thenReturn(it) else Flux.just(it)
```

Comment:
cleanup was being called for every flux item returned, we only want it called to clean up at the end